lib/api/core/storage/indexCache.js

```diff
@@ -0,0 +1,213 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const
+  errorsManager = require('../../../config/error-codes/throw'),
+  Bluebird = require('bluebird');
+
+/**
+ * @deprecated
+ * Index/collection cache management
+ */
+class IndexCache {
+
+  constructor (kuzzle) {
+    this.kuzzle = kuzzle;
+
+    this.internalStorage = this.kuzzle.services.internalStorage;
+    this.publicStorage = this.kuzzle.services.publicStorage;
+
+    // RAM cache structure:
+    //   myIndex: {
+    //     scope: 'public',
```

Comment:
Not with the actual structure because we store the "plain" index and collection names and not the "computed" one.  
The only class aware of this computed name is the Elasticsearch services, core components only knows that they are `public` and `internal` indexes.